src/components/Header.js

```diff
@@ -30,6 +30,8 @@ const exitBtnStyle = {
   right: 0,
 };
 
+let shouldRender = false;
```

Comment:
This should be part of the component state. Then whenever you need to change it you should use this.setState({}). Having anything outside of the class that isn't a const can cause issues.